sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala

```diff
@@ -894,10 +893,6 @@ trait ScalaReflection extends Logging {
 
   import universe._
 
-  // The Predef.Map is scala.collection.immutable.Map.
-  // Since the map values can be mutable, we explicitly import scala.collection.Map at here.
-  import scala.collection.Map
-
```

Comment:
Other changes look fine but this one (https://github.com/apache/spark/commit/05308426f0f51273be95fb1ca2cb1ec19d83cec8). Do you know why the test work now without this fix?

Reply:
Oh, hm, not sure we can remove this. This feels like a Scala 2.13 thing.

Reply:
![image](https://user-images.githubusercontent.com/1475305/99338505-48ca1580-28bf-11eb-9d03-f988de2de689.png)

Reply:
@HyukjinKwon @srowen  seems to be to specify the `Map` type in the `convertToCatalyst` method to `s.c.Map`, but `convertToCatalyst` method is no longer in `ScalaReflection.scala`.

Reply:
That looks like it's on purpose; this is meant to be a supertype of all Maps. It is probably redundant in 2.12 but not 2.13.

Reply:
@srowen seems redundant in Scala 2.13 too,  it is also marked as "unused imports" in IntelliJ, 

It seems that the cases related to 0530842 are as follows:

https://github.com/apache/spark/blob/5af5aa146ecbff38b809127b5eb9805441627ed2/sql/hive/src/test/scala/org/apache/spark/sql/hive/InsertSuite.scala#L105-L120

let me manually test the test this and sql/catalyst,  sql/core in Scala 2.13.

Reply:
@srowen @HyukjinKwon  
Actually there are 2  `import scala.collection.Map` in `ScalaReflection.scala`, one is 

https://github.com/apache/spark/blob/5af5aa146ecbff38b809127b5eb9805441627ed2/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala#L50-L63

the other one is 

https://github.com/apache/spark/blob/5af5aa146ecbff38b809127b5eb9805441627ed2/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/ScalaReflection.scala#L888-L899

this pr removed the `import scala.collection.Map` in the `trait ScalaReflection` not in the `object ScalaReflection`.

Reply:
And after https://github.com/apache/spark/pull/12149, the `import scala.collection.Map` in `trait ScalaReflection` become useless because the last method to used `Map`(`def schemaFor(tpe: `Type`): Schema`) has been moved from `trait ScalaReflection` to `object ScalaReflection`

Reply:
Manual test `SPARK-4052: scala.collection.Map as value type of MapType` , sql/catalyst module , sql/core module  in Scala 2.13 passed

Reply:
OK I get it, seems fine